### PR TITLE
add manual interface for borrowing a GuestPtr

### PR DESCRIPTION
**Pull request by pchickey**
Alex pointed out this issue here: https://github.com/kubkon/wiggle/pull/39#discussion_r389212980

### CODE REVIEW

**Review by alexcrichton — COMMENTED** [review]
Looks good! I think we'll also need to add a `borrow_slice` and `borrow_str` method which work with `GuestPtr<[T]>` and `GuestPtr<str>` too, since the slice specifying the iovecs needs to be added I believe.

**Comment by pchickey** [comment]
Ah good point. I was thinking that was covered by `as_raw` but you may need to borrow something you're just making a copy of as well, in case of overlaps.

File: tests/wasi.rs [hunk 1]
```diff
@@ -130,10 +130,13 @@ impl crate::wasi_snapshot_preview1::WasiSnapshotPreview1 for WasiCtx {
         let mut bc = GuestBorrows::new();
         let mut slices: Vec<&'_ mut [u8]> = Vec::new();
         for iov_ptr in iovs.iter() {
```
Comment (alexcrichton) [inline tests/wasi.rs]: I think just before this you'll want `bc.borrow_slice(iovs)`, but otherwise looks good!

**Review by alexcrichton — APPROVED** [review]

File: tests/wasi.rs (hunk 1 above)
Comment (pchickey) [reply tests/wasi.rs]: Done, thanks

**Pull request merged**